src/hotspot/cpu/aarch64/aarch64_vector.ad

```diff
@@ -6711,3 +6723,38 @@ instruct vexpandBits(vReg dst, vReg src1, vReg src2) %{
   %}
   ins_pipe(pipe_slow);
 %}
+
+// --------------------------------SelectFromTwoVector -----------------------------
+
+instruct vselect_from_two_vectors_SIFNeon(vReg dst, vReg_V17 src1, vReg_V18 src2,
+                                          vReg index, vReg tmp1, vReg tmp2) %{
+  predicate((Matcher::vector_element_basic_type(n) == T_SHORT ||
+            type2aelembytes(Matcher::vector_element_basic_type(n)) == 4) &&
+            (UseSVE < 2 || Matcher::vector_length_in_bytes(n) < 16));
+  match(Set dst (SelectFromTwoVector (Binary index src1) src2));
+  effect(TEMP_DEF dst, TEMP tmp1, TEMP tmp2);
+  format %{ "vselect_from_two_vectors_SIF $dst, $src1, $src2, $index\t# vector (4S/8S/2I/4I/2F/4F). KILL $tmp1, $tmp2" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    uint length_in_bytes = Matcher::vector_length_in_bytes(this);
+    __ select_from_two_vectors_SIFNeon($dst$$FloatRegister, $src1$$FloatRegister,
+                                       $src2$$FloatRegister,$index$$FloatRegister,
+                                       $tmp1$$FloatRegister, $tmp2$$FloatRegister,
+                                       bt, length_in_bytes);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vselect_from_two_vectors(vReg dst, vReg_V17 src1, vReg_V18 src2, vReg index) %{
```

Comment:
Could you please add comment before the rule why `v17` and `v18` are used explicitly here?

Reply:
I'm still curious.

Reply:
Hi @theRealAph , apologies for the late response. The tbl instruction needs both the source registers to be consecutive and I could not find a way to make the register allocator choose two consecutive registers for this operation and decided to hard code them. As v0-v7 are used for function arguments, v8-v15 are non-volatile which are not needed for this purpose (as we dont want to be preserving these values across function calls), I chose two of the volatile registers from v16-v31 for the source registers. Please let me know if this is the right way to approach.

Reply:
I suppose it is, yes. Thanks.